src/InputGroup/_InputGroupAddon.js

```diff
@@ -0,0 +1,49 @@
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const InputGroupAddon = ({
+    children,
+    className,
+    compact,
+    isButton,
+    ...otherProps
+}) => {
+
+    const addonClasses = classnames(
+        'fd-input-group__addon',
+        { 'fd-input-group__addon--button': isButton }
+    );
+
+    return (
+        <span className={addonClasses} {...otherProps} >
+            {React.Children.toArray(children).map(child => {
+                if (!isButton) {
+                    return child;
+                }
+
+                return React.cloneElement(child, {
+                    compact,
+                    className: 'fd-input-group__button'
+                });
+            })}
+        </span>
+    );
+};
+
+InputGroupAddon.displayName = 'InputGroup.Addon';
+
+InputGroupAddon.propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+    compact: PropTypes.bool,
+    isButton: PropTypes.bool
+};
+
+InputGroupAddon.propDescriptions = {
```

Comment:
I think the `compact` prop description should be overwritten here since we are accepting `compact` at the `InputGroup` level and the consumer really shouldn't be passing it here (although nothing bad would happen if they did).
```js
compact: '_INTERNAL USE ONLY._',
```